### PR TITLE
Change agent to support OAuth2 auth

### PR DESCRIPTION
**Pull request by j-fuentes**
This makes the agent support OAuth2.

Signed-off-by: Jose Fuentes <jsfuentescastillo@gmail.com>

### CODE REVIEW

**Comment by j-fuentes** [comment]
I have just realised I haven't changed the Makefile to add LDFLAGS to define `clientID`, `clientSecret` and `authServer`.

Will do tomorrow.

**Comment by j-fuentes** [comment]
This is ready for review :)

File: agentmanigest.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+apiVersion: v1
```
Comment (james-w) [inline agentmanigest.yaml:1]: The filename has a typo: -> `agentmanifest.yaml`.

What are these files for?

File: agentconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+schedule: "* * * * *"
```
Comment (charlieegan3) [inline agentconfig.yaml:1]: We already have agent.yaml - might be worth removing that file?

File: agentmanigest.yaml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      kind: "k8s"
+      config:
+        resource-type:
+          resource: pods
+          version: v1
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: agent
```
Comment (charlieegan3) [inline agentmanigest.yaml:31]: There is a reference to this service account here: https://github.com/jetstack/preflight/blame/master/docs/datagatherers/gke.md#L104

Now seems as good a time as any to correct it from `default`

File: agentmanigest.yaml (hunk 1 above)
Comment (charlieegan3) [inline agentmanigest.yaml:1]: At the moment, these files are documented in the readme: https://github.com/jetstack/preflight#installation

File: pkg/client/client.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/jetstack/preflight/api"
+)
+
+// This variables are injected at build time.
```
Comment (charlieegan3) [inline pkg/client/client.go:13]:
```suggestion
// These variables are injected at build time.
```

File: agentmanigest.yaml (hunk 1 above)
Comment (j-fuentes) [reply agentmanigest.yaml:1]: Ooops, `agentmanigest.yaml` and `agentconfig.yaml` should not be here. They are part of a manual test I was doing and I added them by mistake. Going to delete them right now.

Comment (j-fuentes) [reply agentmanigest.yaml:1]: I added this file by mistake.

File: agentconfig.yaml (hunk 2 above)
Comment (j-fuentes) [reply agentconfig.yaml:1]: I added that file by mistake. It should be deleted now.

**Comment by j-fuentes** [comment]
> This mostly all looks ready to go. ~Do~ Should we update the example manifests to mount in service account credentials as part of this PR?

Not yet, as the production auth server still needs to be configured. Will do later.

File: pkg/client/client.go (hunk 4 above)
Comment (j-fuentes) [reply pkg/client/client.go]: Changed in a new commit (required signoff).

**Pull request merged**